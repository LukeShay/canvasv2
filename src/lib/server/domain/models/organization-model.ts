/* eslint-disable @typescript-eslint/no-var-requires */
/* eslint-disable global-require */
import { Model, RelationMappings, JSONSchema } from 'objection';
import Constants from '../constants';

export class OrganizationModel extends Model {
  static tableName = Constants.ASSISTANTS_TABLE;

  static idColumn = 'id';

  static get jsonSchema(): JSONSchema {
    return {
      properties: {
        addresss1: { type: 'string' },
        addresss2: { type: ['string', 'null'] },
        city: { type: 'string' },
        email: { type: 'string' },
        name: { type: 'string' },
        phoneNumber: { type: 'string' },
        stateId: { type: 'string' },
        website: { type: 'string' },
        zip: { type: 'string' },
      },
      required: ['address1', 'city', 'email', 'name', 'phoneNumber', 'stateId', 'website', 'zip'],
      type: 'object',
    };
  }

  static get relationMappings(): RelationMappings {
    // ts-expect-error
    const { StateModel } = require('./state-model');

    return {
      state: {
        join: {
          from: `${OrganizationModel.tableName}.stateId`,
          to: `${StateModel.tableName}.${StateModel.idColumn}`,
        },
        modelClass: StateModel,
        relation: Model.BelongsToOneRelation,
      },
    };
  }
}
