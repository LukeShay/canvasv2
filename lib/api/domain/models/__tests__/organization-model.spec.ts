import { OrganizationModel } from '../organization-model';

describe('OrganizationModel', () => {
  it("should have idColum 'id'", () => {
    expect(OrganizationModel.idColumn).toEqual('id');
  });

  it('should should have jsonSchema that matches snapshot', () => {
    expect(OrganizationModel.jsonSchema).toMatchSnapshot();
  });

  it('should should have relationMappings that matches snapshot', () => {
    expect(OrganizationModel.relationMappings).toMatchSnapshot();
  });
});
