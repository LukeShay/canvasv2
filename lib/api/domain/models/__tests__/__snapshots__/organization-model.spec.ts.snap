// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizationModel should should have jsonSchema that matches snapshot 1`] = `
Object {
  "properties": Object {
    "addresss1": Object {
      "type": "string",
    },
    "addresss2": Object {
      "type": Array [
        "string",
        "null",
      ],
    },
    "city": Object {
      "type": "string",
    },
    "email": Object {
      "type": "string",
    },
    "name": Object {
      "type": "string",
    },
    "phoneNumber": Object {
      "type": "string",
    },
    "stateId": Object {
      "type": "string",
    },
    "website": Object {
      "type": "string",
    },
    "zip": Object {
      "type": "string",
    },
  },
  "required": Array [
    "address1",
    "city",
    "email",
    "name",
    "phoneNumber",
    "stateId",
    "website",
    "zip",
  ],
  "type": "object",
}
`;

exports[`OrganizationModel should should have relationMappings that matches snapshot 1`] = `
Object {
  "state": Object {
    "join": Object {
      "from": "Assistants.stateId",
      "to": "States.id",
    },
    "modelClass": [Function],
    "relation": [Function],
  },
}
`;
